strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/repository/group/BaseMavenGroupRepositoryComponentTest.java

```diff
@@ -93,63 +93,64 @@ public static void cleanUp()
+        MutableMavenRepositoryConfiguration configuration = new MutableMavenRepositoryConfiguration();
         configuration.setIndexingEnabled(true);
 
-        repository.setStorage(configurationManager.getConfiguration().getStorage(STORAGE0));
         repository.setLayout(Maven2LayoutProvider.ALIAS);
         repository.setAllowsForceDeletion(true);
         repository.setPolicy(RepositoryPolicyEnum.RELEASE.getPolicy());
         repository.setRepositoryConfiguration(configuration);
 
-        super.createRepository(repository);
+        super.createRepository(repository, storageId);
     }
 
-    protected void createLeaf(String repositoryId)
+    protected void createLeaf(String repositoryId,
+                              String storageId)
             throws Exception
     {
-        Repository repository = new Repository(repositoryId);
+        MutableRepository repository = new MutableRepository(repositoryId);
         repository.setLayout(Maven2LayoutProvider.ALIAS);
         repository.setType(new Random().nextInt(2) % 2 == 0 ?
                            RepositoryTypeEnum.HOSTED.getType() :
                            RepositoryTypeEnum.PROXY.getType());
 
-        createRepository(repository);
+        createRepository(repository, storageId);
     }
 
-    protected Repository createGroup(String repositoryId,
-                               String... leafs)
+    protected MutableRepository createGroup(String repositoryId,
+                                            String storageId,
+                                            String... leafs)
             throws Exception
     {
-        Repository repository = new Repository(repositoryId);
+        MutableRepository repository = new MutableRepository(repositoryId);
         repository.setLayout(Maven2LayoutProvider.ALIAS);
         repository.setType(RepositoryTypeEnum.GROUP.getType());
         repository.setGroupRepositories(Sets.newLinkedHashSet(Arrays.asList(leafs)));
 
-        createRepository(repository);
-        
+        createRepository(repository, storageId);
+
         return repository;
     }
 
     @Before
     public void initialize()
             throws Exception
     {
-        createLeaf(REPOSITORY_LEAF_E);
-        createLeaf(REPOSITORY_LEAF_L);
-        createLeaf(REPOSITORY_LEAF_Z);
-        createLeaf(REPOSITORY_LEAF_D);
-        createLeaf(REPOSITORY_LEAF_G);
-        createLeaf(REPOSITORY_LEAF_K);
-
-        createGroup(REPOSITORY_GROUP_C, REPOSITORY_LEAF_E, REPOSITORY_LEAF_Z);
-        createGroup(REPOSITORY_GROUP_B, REPOSITORY_GROUP_C, REPOSITORY_LEAF_D, REPOSITORY_LEAF_L);
-        createGroup(REPOSITORY_GROUP_A, REPOSITORY_LEAF_G, REPOSITORY_GROUP_B);
-        createGroup(REPOSITORY_GROUP_F, REPOSITORY_GROUP_C, REPOSITORY_LEAF_D, REPOSITORY_LEAF_L);
-        createGroup(REPOSITORY_GROUP_H, REPOSITORY_GROUP_F, REPOSITORY_LEAF_K);
+        createLeaf(REPOSITORY_LEAF_E, STORAGE0);
```

Comment:
Shouldn't this `createLeaf()` method's parameters be the other way around -- `storageId` and `repositoryId` -- as that's the logical ordering, right...? (You can sort this out in a follow-up pull, if you agree).